### PR TITLE
Confetti effect when a restaurant is drawn

### PR DESCRIPTION
**Pull request by otsha**
- Added [p5.js](https://p5js.org) as a dependency
- When a restaurant result is shown, a 3-second confetti pop animation is played
  - When the animation is done playing, the canvas instance is destroyed
- Basic tests: the component is not shown when a restaurant has not yet been drawn and vice versa

### CODE REVIEW

File: packages/app/src/components/Confetti/Confetti.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react'
+import confettiScript from '../../services/confetti'
```
Comment (ritinae) [inline packages/app/src/components/Confetti/Confetti.js:2]: Importing `*Script` from `services/` seems a bit iffy

File: packages/app/src/components/Confetti/Confetti.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.confetti-canvas > .p5Canvas {
+    position: absolute !important;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    z-index: 10 !important;
+    pointer-events: none;
+}
```
Comment (ritinae) [inline packages/app/src/components/Confetti/Confetti.css:8]: T R I G G E R E D
```suggestion
}

```

**Review by skoskipaa — COMMENTED** [review]
Code looks ok (except for the famous missing line from .css file...)

File: packages/app/src/services/confetti.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import p5 from 'p5'
+
+/**
+ * A p5.js 'sketch' that renders the confetti pop animation.
+ * Further information about p5: https://p5js.org
+ * @param {*} sketch
```
Comment (ritinae) [inline packages/app/src/services/confetti.js:6]: If you provide JSDoc comment with a `@param` tag, please add a parameter description too, e.g. like this:
```
 * @param {*} sketch a short explanation on what the parameter is
```
The syntax is
```
@param {ParameterType} parameterName Description text
```
If no description nor the type are provided, the `@param` tag is useless and should be removed (or more information added).

For further information, see [`@param`](https://jsdoc.app/tags-param.html)

File: packages/app/src/services/confetti.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+   * Init an array of confetti particles with:
+   * - The shape of the particle
+   * - A position (as a vector)
+   * - A target position / direction (as a vector)
+   * - A random color @see randomColor
+   * - Angle
+   * - Quasi-random rotation speed
+   */
+
+  const initParticles = () => {
```
Comment (ritinae) [inline packages/app/src/services/confetti.js:29]: `@see` tags are not intended to be used like this. They should be at the end of the JSDoc comment, like this:
```suggestion
  /**
   * Init an array of confetti particles with:
   * - The shape of the particle
   * - A position (as a vector)
   * - A target position / direction (as a vector)
   * - A random color
   * - Angle
   * - Quasi-random rotation speed
   *
   * @see randomColor
   */
  const initParticles = () => {
```
If linking in-line is desired, `{@link targetName}` can be used.

For further reference: [`@link`](https://jsdoc.app/tags-inline-link.html), [`@see`](https://jsdoc.app/tags-see.html)

File: packages/app/src/services/confetti.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      opacity -= 4
+    }
+  }
+
+  /**
+   * Draws a singular confetti particle
+   * @param {*} p - The particle to be drawn
+   */
+
+  const drawParticle = (p) => {
```
Comment (ritinae) [inline packages/app/src/services/confetti.js:105]: JSDoc comments are part of the function definition and thus should not be separated from it by whitespace.

**Review by ritinae — APPROVED** [review]
Pointed out a few non-issues.

**Pull request merged**